Libs/RaptureCore/src/main/java/rapture/dp/DefaultDecisionProcessExecutor.java

```diff
@@ -522,6 +525,39 @@ private void markAsFinished(final WorkOrder workOrder, final Worker worker, Work
             // remove the completed id from the master list
             ids.remove(id);
             workOrder.setPendingIds(ids);
+            
+            // Clear up temporary output storage
+            BlobApi blobApi = Kernel.getBlob();
+            String workOrderURI = worker.getWorkOrderURI();
+            String outputUri = (workOrderURI.startsWith("//")) 
+            		? Scheme.BLOB+":"+workOrderURI
+                	: Scheme.BLOB + (workOrderURI.substring(workOrderURI.indexOf(':')));
```

Comment:
Perhaps this construct (you use it earlier as well) could be a static helper function on RaptureURI